micropip/transaction.py

```diff
@@ -335,22 +338,42 @@ def find_wheel(metadata: ProjectInfo, req: Requirement) -> WheelInfo:
             )
             continue
 
-        best_wheel = None
-        best_tag_index = float("infinity")
+        wheels = list(releases[ver])
+
+        # If the version is yanked, put it in the end of the candidate list.
+        # If we can't find a wheel that satisfies the requirement,
+        # install the yanked version as a last resort.
+        yanked = any(wheel.yanked for wheel in wheels)
+        if yanked:
+            yanked_versions.append(wheels)
+            continue
```

Comment:
I think this control flow makes things a bit confusing for me, coupled with the fact that the variable defined above is named `wheels`. I understand that we are trying a two-pass approach by first looking for non-yanked wheels, and then searching within the yanked wheels if no non-yanked wheel in the first pass was found to be a suitable candidate for installation.

Instead of checking if a yanked wheel exists, we could create a smaller list using a list comprehension and manage its lifetime separately instead of branching out? i.e., using `yanked_wheels = [wheel for wheel in wheels if wheel.yanked]` and then passing that to `_find_best_wheel()` (which can return `None` if it receives `None` as an input).

Reply:
Thanks for the explanation! This makes more sense to me. So, if I understood it correctly, we are checking for the version first and finding wheels from it (and then checking which is yanked) – rather than the other way around.

Reply:
Yes, exactly.